javagen/src/main/java/com/azure/autorest/template/ProtocolSampleTemplate.java

```diff
@@ -79,6 +79,7 @@ public void write(ProtocolExample protocolExample, JavaFile javaFile) {
             boolean matchRequiredParameter = false;
             for (int i = 0; i < numParam; i++) {
                 ClientMethodParameter p = method.getParameters().get(i);
+                // TODO: should use getRequestParameterName from proxy method parameter, instead of getName from client method parameter
                 if (p.getName().equalsIgnoreCase(parameterName)) {
```

Comment:
update for this TODO as well?

Reply:
Probably going to let you or Xiaofei to improve it :-)

Reply:
https://github.com/Azure/autorest.java/issues/1313

Reply:
sure, I can do that